src/Microsoft.IdentityModel.Tokens/Validators.cs

```diff
@@ -38,6 +38,34 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     public static class Validators
     {
+        /// <summary>
+        /// Validates if a given algorithm for a <see cref="SecurityKey"/> is acceptable
+        /// </summary>
+        /// <param name="validationParameters"><see cref="TokenValidationParameters"/> required for validation.</param>
+        /// <param name="algorithm">The algorithm being considered if it is valid</param>
+        /// <param name="securityKey">The <see cref="SecurityKey"/> that signed the <see cref="SecurityToken"/>.</param>
+        /// <param name="securityToken">The <see cref="SecurityToken"/> being validated.</param>
+        /// <returns><c>true</c> if the algorithm is acceptable</returns>
+        public static bool ValidateAcceptedAlgorithms(TokenValidationParameters validationParameters, string algorithm, SecurityKey securityKey, SecurityToken securityToken)
+        {
+            if (validationParameters == null)
+                throw LogHelper.LogArgumentNullException(nameof(validationParameters));
+
+            if (validationParameters.AcceptedAlgorithms != null && validationParameters.AcceptedAlgorithms.Any() && !validationParameters.AcceptedAlgorithms.Contains(algorithm, StringComparer.Ordinal))
+            {
+                LogHelper.LogInformation(LogMessages.IDX10696, algorithm);
+                return false;
+            }
+
+            if (validationParameters.AcceptedAlgorithmValidator != null && !validationParameters.AcceptedAlgorithmValidator(algorithm, securityKey, securityToken, validationParameters))
```

Comment:
AcceptedAlgorithmValidator delegate should have priority over validation of AcceptedAlgorithms.